vocode/streaming/agent/state_agent.py

```diff
@@ -1350,6 +1354,14 @@ async def call_ai(self, prompt, tool=None, stop=None, stream_output=True):
                     response_text += text_chunk
                     if any(token in text_chunk for token in stop_tokens):
                         break
+            # For non-streaming, update average latency at completion
+            end_time = time.time()
+            if not hasattr(self, "average_latency"):
+                self.average_latency = end_time - start_time
+            else:
+                self.average_latency = (
+                    self.average_latency + (end_time - start_time)
+                ) / 2  # units are in
```

Comment:
why is this over 2? if you want a rolling average of latency, the denominator is determined by the number of messages

I recommend replacing `self.average_latency` with an array of latencies, and computing the average from scratch when you need it. We can wrap that logic in helper functions and caching down the line, as necessary